deepsubtree.go

```diff
@@ -16,6 +16,83 @@ type DeepSubTree struct {
 	*MutableTree
 }
 
+type DSTNonExistenceProof struct {
+	*ics23.NonExistenceProof
+	leftSiblingProof  *ics23.ExistenceProof
+	rightSiblingProof *ics23.ExistenceProof
+}
+
+func convertToDSTNonExistenceProof(
+	tree *MutableTree,
+	nonExistenceProof *ics23.NonExistenceProof,
+) (*DSTNonExistenceProof, error) {
+	dstNonExistenceProof := DSTNonExistenceProof{
+		NonExistenceProof: nonExistenceProof,
+	}
+	if nonExistenceProof.Left != nil {
+		leftSibling, err := tree.GetSiblingNode(nonExistenceProof.Left.Key)
+		if err != nil {
+			return nil, err
+		}
+		dstNonExistenceProof.leftSiblingProof, err = tree.createExistenceProof(leftSibling.key)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if nonExistenceProof.Right != nil {
+		rightSibling, err := tree.GetSiblingNode(nonExistenceProof.Right.Key)
+		if err != nil {
+			return nil, err
+		}
+		dstNonExistenceProof.rightSiblingProof, err = tree.createExistenceProof(rightSibling.key)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &dstNonExistenceProof, nil
+}
+
+func (tree *ImmutableTree) GetSiblingNode(key []byte) (*Node, error) {
```

Comment:
Nit: Document public methods!

Reply:
Also, exported method is not tested.

Reply:
Converted `ConvertToDSTNonExistenceProof` to an exported method instead and documented it.

Reply:
> Nit: Document public methods!
Shouldn't be a nit, should be a standard :-)